src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -191,11 +252,24 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
 
         List<ProgrammingSubmission> submissions = programmingExerciseService.notifyChangedTestCases(exerciseId, requestBody);
 
-        // notify users via websocket.
+        notifyUserTriggerBuildForNewSubmissions(submissions);
+
+        return ResponseEntity.ok().build();
+    }
+
+    private void notifyUserTriggerBuildForNewSubmissions(Collection<ProgrammingSubmission> submissions) {
         for (ProgrammingSubmission submission : submissions) {
-            messagingTemplate.convertAndSend("/topic/participation/" + submission.getParticipation().getId() + "/newSubmission", submission);
+            notifyUserAndTriggerBuildForNewSubmission(submission);
```

Comment:
Did you test this for a couple of hundred of students? Or even thousands?
Please talk to @sleiss 
He has a script to generate up to 500 participations with existing build results with the 500 artemis_test_users that we have. If we need more, we can create more test users

Reply:
I will discuss it with @sleiss, but I think we can merge this PR when it's done regardless as the feature is pretty isolated from the general submission process.

Reply:
Sounds good, I agree, we can still test this after the merge. Would just be interesting how the performance characteristics of the for loop are with many users.